File: Data Reverse.py
def data_reverse(data):
    result = []
    #function to split list into chunks
    def chunks(data):
        for i in range(0, len(data), 8):
            yield data[i:i+8]
    
    #split into chunks and reverse it
    a = list(chunks(data))[::-1]
    
    b = ""
    
    #add the nested list to b
    for i in a:
        for x in i:
            b += str(x)
    
    #convert the string elements into int and append to result
    for i in b:
        result.append(int(i))
    
    return result
    

File: Data Reverse.js
function dataReverse(data) {
  // Your code here
  
  //need to split the array into chunks of 8
  var a = data.join("").match(/.{1,8}/g)
  var c = []
  var result = []
  
  if(data.length = 0 || data == null){
    return []
  } else {
    //reverse the order of the chunks and push into c
    for(let i = a.length-1; i >= 0; i--){
      c.push(a[i])
    }
    
    c = c.join("") //join them, but as a string??
    
    //put them into result as integers!!
    for(let i = 0; i < c.length; i++){
      result.push(parseInt(c[i]))
    }
    
    return result
  }
}
